File: 03_A/VB/CSV_Reader/DiscreteDistribution.vb
Public Class DiscreteDistribution
    Private distDict As SortedDictionary(Of Object, Integer)
    Private n As Integer

    Public Sub New()
        distDict = New SortedDictionary(Of Object, Integer)()
        n = 0
    End Sub

    Public Sub AddValue(ByVal key As Object)
        If distDict.ContainsKey(key) Then
            distDict(key) += 1
        Else
            distDict.Add(key, 1)
        End If

        n += 1
    End Sub

    Public Function GetRelFreq(ByVal key As Object) As Double
        Dim relFreq = distDict(key) / n
        Return relFreq
    End Function

    Public Function GetPerc(ByVal key As Object) As Double
        Dim perc = distDict(key) / n * 100
        Return perc
    End Function

    Public ReadOnly Property Distribution As SortedDictionary(Of Object, Integer)
        Get
            Return distDict
        End Get
    End Property
End Class

File: StatisticsLibrary/StatisticsLib/DiscreteDistribution.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StatisticsLib {
    public class DiscreteDistribution {
        private SortedDictionary<object, int> distDict;
        private int n;

        public DiscreteDistribution() {
            distDict = new SortedDictionary<object, int>();
            n = 0;
        }

        public void AddValue(object key) {
            if (distDict.ContainsKey(key)) {
                distDict[key]++;
            } else {
                distDict.Add(key, 1);
            }
            n++;
        }

        public int GetCount(object key) => distDict[key];

        public double GetRelFreq(object key) {
            double relFreq = distDict[key] / (double)n;
            return relFreq;
        }

        public double GetPerc(object key) {
            double perc = distDict[key] / (double)n * 100;
            return perc;
        }

        public SortedDictionary<object, int> Distribution { get => distDict; }


    }
}
